### DISCUSSION TITLE
Know about the system Requirements

### DISCUSSION THREAD

**Discussion by monesh9732** [evershopcommerce/evershop#576] [Q&A]
Hi Mates, 

This framework has a large code base, allowing us to develop and structure an entire e-commerce site. It seamlessly integrates both backend and frontend applications within a single framework. However, I need to know the RAM and ROM requirements for this framework.

When running the application on an Ubuntu server with 2 GB RAM and 2 GB swap memory, I encounter significant memory load issues. Additionally, the npm run build command takes an excessive amount of time to build the application. Running the site with the npm run dev command results in very slow performance.

Please help me resolve these issues. Thank you in advance.

**Comment by treoden** [comment]
Hi @monesh9732 ,

Let me explain a little bit about why `npm run dev` is slow:

In the development mode, your frontend code will be built on run time. It means when you hit a request, EverShop will collect all components and build your frontend code. EverShop is multi page application, meaning it will have to do this process for every page you visited.

In production mode, you need to run the `npm run build` first, this command will build your production version and that is why the production mode is very fast.

Thanks
